Microsoft.DotNet.Interactive/KernelExtensionPathResolver.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using MLS.Agent.Tools;
+
+namespace Microsoft.DotNet.Interactive
+{
+    public class KernelExtensionPathResolver
+    {
+        public static bool TryGetExtensionPath(DirectoryInfo directory, IKernel handlingKernel, out DirectoryInfo extensionDirectory)
+        {
+            if(handlingKernel.Name == "csharp")
```

Comment:
It's possible to name a `CSharpKernel` differently, so this would be more robust if you check the type of the kernel.

Reply:
I am running into the layering problem again. The Csharp kernel lives in the WorkspaceServer and we cant reference it from the `Microsoft.Dotnet.Interactive` assembly.

Reply:
One possible approach then might be to ask the `IKernel` what its path convention is instead of hard-coding it. This could be done by adding an interface, something like `IExtensibleKernel`, because not all `IKernel`s might support extensions via NuGet. 

(I think we'll also want to move `CSharpKernel` into a new project, `Microsoft.DotNet.Interactive.CSharp`, but that doesn't have to happen in this PR).